Common/Python/PandasConverter.cs

```diff
@@ -61,149 +61,271 @@ public PandasConverter(PyObject pandas = null)
+        public void Add(IBaseData baseData)
+        {
+            _timeIndex.Add(baseData.Time);
+
+            var bar = baseData as IBaseDataBar;
+            if (bar != null)
             {
-                tuples = time.Select(x => new PyTuple(new PyObject[] { symbol.ID.Date.ToPython(), symbol.ID.StrikePrice.ToPython(), symbol.ID.OptionRight.ToString().ToPython(), value, x.ToPython() }));
-                names = "expiry,strike,type," + names;
+                _series["open"].Add((double)bar.Open);
+                _series["high"].Add((double)bar.High);
+                _series["low"].Add((double)bar.Low);
+                _series["close"].Add((double)bar.Close);
+
+                var tradeBar = bar as TradeBar;
+                if (tradeBar != null)
+                {
+                    _series["volume"].Add((double)tradeBar.Volume);
+                }
+
+                var quoteBar = bar as QuoteBar;
+                if (quoteBar != null)
+                {
+                    _series["asksize"].Add((double)quoteBar.LastAskSize);
+                    _series["bidsize"].Add((double)quoteBar.LastBidSize);
+
+                    if (quoteBar.Ask != null)
+                    {
+                        _series["askopen"].Add((double)quoteBar.Ask.Open);
+                        _series["askhigh"].Add((double)quoteBar.Ask.High);
+                        _series["asklow"].Add((double)quoteBar.Ask.Low);
+                        _series["askclose"].Add((double)quoteBar.Ask.Close);
+                    }
+                    else
+                    {
+                        _series["askopen"].Add(double.NaN);
+                        _series["askhigh"].Add(double.NaN);
+                        _series["asklow"].Add(double.NaN);
+                        _series["askclose"].Add(double.NaN);
+                    }
+
+                    if (quoteBar.Bid != null)
+                    {
+                        _series["bidopen"].Add((double)quoteBar.Bid.Open);
+                        _series["bidhigh"].Add((double)quoteBar.Bid.High);
+                        _series["bidlow"].Add((double)quoteBar.Bid.Low);
+                        _series["bidclose"].Add((double)quoteBar.Bid.Close);
+                    }
+                    else
+                    {
+                        _series["bidopen"].Add(double.NaN);
+                        _series["bidhigh"].Add(double.NaN);
+                        _series["bidlow"].Add(double.NaN);
+                        _series["bidclose"].Add(double.NaN);
+                    }
+                }
             }
 
-            return _pandas.MultiIndex.from_tuples(tuples.ToArray(), names: names.Split(','));
+            var data = baseData as DynamicData;
+            if (data != null)
+            {
+                foreach (var kvp in data.GetStorageDictionary())
+                {
+                    _series[kvp.Key].Add(Convert.ToDouble(kvp.Value));
```

Comment:
I think we want the value property also for custom data types, such as Bitcoin

Reply:
That was included in the line below. :)